src/test/java/com/ncc/aif/ExamplesAndValidationTest.java

```diff
@@ -1359,9 +1369,45 @@ void invalid() {
 
             @Test
             void valid() {
+                // setup() already creates an entity, relation, and event in clusters with an IRI.
                 testValid("NIST.valid: Cluster has IRI");
             }
         }
+
+        // Each entity/relation/event type statement must have at least one justification
+        @Nested
+        class JustifyTypeAssertions {
+            @Test
+            void invalid() {
+                // Create an entity, but do not mark its type assertion with a justification.
+                final Resource newEntity = makeEntity(model, getEntityUri(), system);
+                addType(newEntity, SeedlingOntology.Person);
+                makeClusterWithPrototype(model, getClusterUri(), newEntity, system);
+
+                // Create an event, but do not mark its type assertion with a justification.
+                final Resource newEvent = makeEvent(model, getUri("event-5"), system);
+                addType(newEvent, SeedlingOntology.Conflict_Attack);
+                makeClusterWithPrototype(model, getClusterUri(), newEvent, system);
+
+                // Create a relation, but do not mark its type assertion with a justification.
+                final Resource newRelation = makeRelation(model, getUri("relation-2"), system);
+                addType(newRelation, SeedlingOntology.GeneralAffiliation_APORA);
+                makeClusterWithPrototype(model, getClusterUri(), newRelation, system);
+
+                testInvalid("NIST.invalid: type assertions must be justified");
+            }
+
+            @Test
+            void valid() {
+                // setup() already already makes type assertions on the entity, event,
+                // and relation objects, justified by a text justification.
+
+                // Note that if you use makeRelationInEventForm, you will need to use the Jena API to obtain
```

Comment:
I don't know-- is `makeRelationInEventForm()` just generally bad/outdated/useless/OBE/etc?  Or is it just an issue for restricted AIF because of this restriction?  We don't use it in VERDI, but it's used in several of the examples...it seems we should remove it from the examples if we're deprecating it.

Reply:
It was originally added to help researcher bridge the gap between the old way of adding relations and the new way. Seeing as they have to do all the requisite steps for events anyway, they should also be doing them for relations. The method is rather monolithic and just wraps 4 other calls. At best, it's a convenience function. As researchers are going to need access to the type statements, it's now more of an in-convenience function. Whenever we decide to deprecate it, we will have to remove it from the examples

Reply:
I've added it to AIDA-698 which talks about refactoring the tests.